src/actions/profile-view.js

```diff
@@ -149,6 +150,46 @@ export function selectLeafCallNode(
   };
 }
 
+/**
+ * Given a threadIndex and a sampleIndex, select the call node at the bottom ("root")
+ * of that sample's stack.
+ */
+export function selectRootCallNode(
+  threadIndex: ThreadIndex,
+  sampleIndex: IndexIntoSamplesTable
+): ThunkAction<void> {
+  return (dispatch, getState) => {
+    const threadSelectors = getThreadSelectors(threadIndex);
+    const filteredThread = threadSelectors.getFilteredThread(getState());
+    const callNodeInfo = threadSelectors.getCallNodeInfo(getState());
+
+    const newSelectedStack = filteredThread.samples.stack[sampleIndex];
+    const newSelectedCallNode =
+      newSelectedStack === null
```

Comment:
I'd suggest to check for this case before and return early. And also to check for `undefined`, just in case the `sampleIndex` is out of bounds.
```js
if (newSelectedStack === null || newSelectedStack === undefined) {
  return;
}
```
Then we don't need to worry about this case below.